src/myrtlespeech/run/run.py

```diff
@@ -47,72 +48,216 @@ def __call__(self, sentence: List[str]) -> List[str]:
+                else:
+                    raise ValueError("error_rate is not in ['cer', 'wer'].")
+
+                self.distances[error_rate].append(distance)
 
     def on_epoch_end(self, **kwargs):
         if self.training:
             return
-        wer = float(sum(self.distances)) / sum(self.lengths) * 100
-        kwargs["reports"][self.ctc_decoder.__class__.__name__]["wer"] = wer
+        for error_rate in self.calc_quantities:
+            lengths = sum(self.lengths[error_rate])
+            if lengths != 0:
+                err = (
+                    float(sum(self.distances[error_rate]))
+                    / sum(self.lengths[error_rate])
+                    * 100
+                )
+            else:
+                warnings.warn(
+                    "Total length of input sequences == 0. \
+                    Cannot calculate WER/CER"
+                )
+                err = -1  # return infeasible value
+
+            kwargs["reports"][self.decoder.__class__.__name__][
+                error_rate
+            ] = err
+
+
+class ReportCTCDecoder(ReportDecoderBase):
+    """CTC Decoder Callback
+
+    Args:
+        See :py:class`ReportDecoderBase`.
+    """
+
+    def __init__(
+        self,
+        ctc_decoder,
+        alphabet,
+        word_segmentor,
+        eval_every=1,
+        calc_quantities=("cer", "wer"),
+    ):
+        super().__init__(
+            ctc_decoder, alphabet, word_segmentor, eval_every, calc_quantities
+        )
+
+    @property
+    def decoder_input_key(self):
+        return "last_output"
+
+
+class ReportRNNTDecoder(ReportDecoderBase):
+    """RNNT Decoder Callback.
+
+    Args:
+        skip_first_epoch: bool. Default = False. If True, the first eval epoch
+            is skipped. This is useful as the decoding is *very* slow with an
+            un-trained model (i.e. inference is considerably faster when the
+            model is more confident in its predictions).
+
+        See :py:class`ReportDecoderBase` for other args.
+    """
+
+    def __init__(
+        self,
+        rnnt_decoder,
+        alphabet,
+        word_segmentor=WordSegmentor(" "),
+        eval_every=1,
+        calc_quantities=("cer", "wer"),
+        skip_first_epoch=False,
+    ):
+        super().__init__(
+            rnnt_decoder, alphabet, word_segmentor, eval_every, calc_quantities
+        )
+        self.skip_first_epoch = skip_first_epoch
+
+    def on_batch_end(self, **kwargs):
+        if self.skip_first_epoch and kwargs["epoch"] == 0:
+            return
+        return super().on_batch_end(**kwargs)
+
+    @property
+    def decoder_input_key(self):
+        return "last_input"  # i.e. we can't use the rnnt output during
+        # decoding as this was computed using ground-truth labels!
 
 
 class TensorBoardLogger(ModelCallback):
-    """TODO"""
+    r"""Enables TensorBoard logging.
+
+    .. note::
+        If :py:class:`MixedPrecision` is used then this should appear earlier
```

Comment:
Add a blank line between the directive name and the text else the text becomes an option, see: 

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#rst-directives